agent/agentbootstrap/bootstrap.go

```diff
@@ -98,6 +100,20 @@ func InitializeState(
 	if !ok {
 		return nil, errors.Errorf("state serving information not available")
 	}
+
+	// The JWKS refresh URL is a public key that we trust for federated
```

Comment:
This isn't a direct `InitializeState` concern and the implementation is incompatible with 4.0. My idea (perhaps unclear) was to do this work as part of the workflow that calls `InitializerState`. ie in `cmd/jujud/agent/bootstrap.go` the `Run` method does all the set up; just add this to that. And using the pattern already established:
```
var (
	initiateMongoServer  = peergrouper.InitiateMongoServer
	agentInitializeState = agentbootstrap.InitializeState
	sshGenerateKey       = ssh.GenerateKey
	minSocketTimeout     = 1 * time.Minute
)
```
the check can be stubbed for a test to be written.

Reply:
I see, done.